consensus/leader.go

```diff
@@ -179,67 +194,94 @@ func (consensus *Consensus) onCommit(msg *msg_pb.Message) {
 		return
 	}
 
-	// let it handle its own logs
-	// TODO(Edgar)(TEMP DISABLE while testing double sign)
-	if !consensus.onCommitSanityChecks(recvMsg) {
+	// NOTE let it handle its own log
+	if !consensus.isRightBlockNumAndViewID(recvMsg) {
 		return
 	}
 
 	consensus.mutex.Lock()
 	defer consensus.mutex.Unlock()
 
+	if key := (bls.PublicKey{}); consensus.couldThisBeADoubleSigner(recvMsg) {
+		if alreadyCastBallot := consensus.Decider.ReadBallot(
```

Comment:
This is just checking whether there is a matching commit in the current round, right? What if the double sign is on a previous block that's already in the chain.